src/utils/tokens.js

```diff
@@ -31,6 +31,9 @@ const tokens = {
   addToken(uid, name, symbol) {
     const tokens = hathorLib.tokens.addToken(uid, name, symbol);
     store.dispatch(newTokens({tokens, uid: uid}));
+    const reduxState = store.getState();
+    const reduxWallet = reduxState.wallet;
+    wallet.fetchTokensMetadata([uid], reduxWallet.conn.network);
```

Comment:
What if it fails? Do we automatically retry? How many times?

Reply:
I've refactored this part of code to have an error handling and auto retry. Can you see if it's good? https://github.com/HathorNetwork/hathor-wallet/pull/228/commits/9e150da7e7f5923df00911378430609f9345f9aa

Reply:
After a second refactor, this logic moved to the hathor-wallet-lib.